python/ray/tune/trainable.py

```diff
@@ -225,10 +228,16 @@ def save(self, checkpoint_dir=None):
                                          "wb"))
         else:
             raise ValueError("Return value from `_save` must be dict or str.")
-        pickle.dump([
-            self._experiment_id, self._iteration, self._timesteps_total,
-            self._time_total, self._episodes_total, saved_as_dict
-        ], open(checkpoint_path + ".tune_metadata", "wb"))
+        pickle.dump({
+            "experiment_id": self._experiment_id,
+            "iteration": self._iteration,
+            "checkpoint_num": self._checkpoint_num,
+            "timesteps_total": self._timesteps_total,
+            "time_total": self._time_total,
+            "episodes_total": self._episodes_total,
+            "saved_as_dict": saved_as_dict
+        }, open(checkpoint_path + ".tune_metadata", "wb"))
+        self._checkpoint_num += 1
```

Comment:
Use self.iteration

Reply:
I'm not sure if that makes sense when you do `checkpoint_at_end`... you get something like `experiment/checkpoint_433.pkl`